### PR TITLE
Add spiking conversion

### PR DESCRIPTION
**Pull request by CodyCBakerPhD**
@bendichter @alejoe91 @weiglszonja @MRScheid This is the branch that I'm using to debug the processing pipeline. Depends on [this branch of nwb-conversion-tools](https://github.com/catalystneuro/nwb-conversion-tools/tree/add_save_si_object_function).

Still a couple things left to do on metadata format, especially for manually passed recordings electrode information. But the `spikeinterface_pipeline_testing.ipynb` contains the essential parts of the main code, but truncates the data at the start to a specifiable % of the total data size (default atm is 2.5%), because otherwise the spike sorting part takes a _very_ long time. But it is working at least on my end (though I'm only able to use ironclust without triggering sorter specific errors right now), and is outputting spiking units (though not very good looking ones, likely needs a bit of fine tuning on the actual data).

With 2.5% of the data the sorting is done within a couple minutes. Also need to compress the dumping part using the magic functions from `nwb-conversion-tools`.

I still use the `stub_test` conversion option just to make the conversion step go faster; @MRScheid if you need more data than is contained in the stub I posted on slack, just run this script with the `stub_test` set to false and you'll get a lot more data (will take maybe 10 minutes or so to write it all I think).

### CODE REVIEW

**Comment by bendichter** [comment]
Cool. We agreed to age out because we didn't trust it

File: tank_lab_to_nwb/__init__.py [hunk 1]
```diff
@@ -1 +1,2 @@
 from .convert_towers_task.towersnwbconverter import TowersNWBConverter
+from .convert_towers_task.tanknwbconverter import TankNWBConverter
```
Comment (weiglszonja) [inline tank_lab_to_nwb/__init__.py]: At some point is this `tanknwbconverter` going to replace `towersnwbconverter`?

Comment (bendichter) [reply tank_lab_to_nwb/__init__.py]: Good point, @weiglszonja. The Tank Lab does more than this experiment, and we should make this robust to potentially having more converters for them in the future, so I'd rather have a single converter and call is something like `TowersNeuropixelNWBConverter`

**Comment by CodyCBakerPhD** [comment]
@bendichter @weiglszonja I went ahead and changed the names on both classes; the one we originally designed is the `TowersRawNWBConverter `and doesn't handle synchronization (since we can't without making somewhat strong assumptions about *how* this particular recording happened to have it's ttl designed). The one for the processing pipeline is now the `TowersProcessedNWBConverter`. Open to other naming conventions as well.

Notebooks now support `save_si_object` function which really reduces the number of code blocks as we approach the final conversion step.

**Pull request merged**